Uncap hardcode HUD

GenericHeroGuy opened the pull request:
As the title suggests, this PR uncaps... uh... everything!

Here's the full list:
* FINISH and LAP animations
* Level title (intro)
* Intermission scoreboard
* Minimap icons
* The title screen
* Input display
* Bottom-right logo in title screen demos
* Battle full-screen animations
* ~~VHS effect~~ misunderstood

Besides that, the other notable changes are:
* LAP animation no longer segfaults on laps greater than 99
* ~~Added `k_oldcardanimation`; the battle animations are tied to the ticrate in a non-trivial way, so I had to interpolate them rather than uncapping 😅 (maybe there's a better way?)~~ nope, breaks [save games](https://github.com/Indev450/SRB2Kart-Saturn/blob/9b7eff1a33e57a2589173d3a8c4f3d2a2d0bd0ae/src/p_saveg.c#L138)
* Fixed nametags in splitscreen

~~TODO: `K_getScreenCoords` still needs uncapping~~ done!

GenericHeroGuy committed:
Uncap FINISH marquee, level title, intermission scoreboard

GenericHeroGuy committed:
Uncap lap animation

Now handles laps greater than 99

GenericHeroGuy committed:
Uncap minimap icons

GenericHeroGuy committed:
Uncap title screen

GenericHeroGuy committed:
Uncap input display and demo title

GenericHeroGuy committed:
Needs more stplyrnum

GenericHeroGuy committed:
Interp- I mean, uh, uncap battle animations

Indev450 commented on src/d_player.h:
```diff
@@ -250,6 +250,7 @@ typedef enum
 	k_lapanimation,		// Used to show the lap start wing logo animation
 	k_laphand,			// Lap hand gfx to use; 0 = none, 1 = :ok_hand:, 2 = :thumbs_up:, 3 = :thumps_down:
 	k_cardanimation,	// Used to determine the position of some full-screen Battle Mode graphics
+	k_oldcardanimation,	// Used to interpolate full-screen Battle Mode graphics
```
This can probably cause desynch with vanilla clients since kartstuff is synched so fields after k_oldcardanimations will be synched wrong

alufolie91 replied:
This breaks netcompability with vanilla hard. Joining a server hosted with this results in the game throwing an I_Error 
![Screenshot (136)](https://github.com/Indev450/SRB2Kart-Saturn/assets/136046486/162025d9-476c-461c-8f35-60d2ff8091ff)

GenericHeroGuy replied:
Shouldn't have rushed that one...
Turns out there *was* a better way after all!

alufolie91 commented:
One other thing, it does change how the game looks and feels visually quite drastically.
Dont know how much extra work it would be, but id prefer if this could be toggable
(sounds kinda redundant ik, but i already see ppl or atleast a certain people complaining about this change)

GenericHeroGuy committed:
Actually uncap battle animations

...am I really using "uncap" and "interpolate" correctly?

GenericHeroGuy committed:
Add cv_uncappedhud toggle and R_GetHudUncap

Turns out I need to AND with FRACMASK everywhere to avoid discrepencies,
so might as well make a function for it

(no, it's NOT interpolation, I swear...!)

GenericHeroGuy committed:
Uncap K_GetScreenCoords

Plus fix nametags in splitscreen

GenericHeroGuy committed:
Merge branch 'Saturn-Next' into hud-uncap

GenericHeroGuy committed:
Uncap VHS effect

GenericHeroGuy committed:
Fix VHS timescale issues

...it was already uncapped in vanilla
